src/app/technical-record/technical-record.container.ts

```diff
@@ -0,0 +1,72 @@
+import { Component, OnInit, ChangeDetectionStrategy } from '@angular/core';
+import { Observable } from 'rxjs';
+import { Store, select } from '@ngrx/store';
+import { IAppState } from '@app/store/state/app.state';
+
+import { selectSelectedVehicleTestResultModel } from '@app/store/selectors/VehicleTestResultModel.selectors';
+import {
+  selectVehicleTechRecordModelHavingStatusAll,
+  getVehicleTechRecordAdrMetaData,
+  getSelectedVehicleTechRecord,
+  getViewState
+} from '@app/store/selectors/VehicleTechRecordModel.selectors';
+import { MetaData } from '@app/models/meta-data';
+import { TechRecord } from '@app/models/tech-record.model';
+import {
+  SetViewState,
+  UpdateVehicleTechRecord
+} from '@app/store/actions/VehicleTechRecordModel.actions';
+import { VehicleTechRecordModel } from '@app/models/vehicle-tech-record.model';
+import { TestResultModel } from '@app/models/test-result.model';
+import { ActivatedRoute } from '@angular/router';
+import { VIEW_STATE } from '@app/app.enums';
+
+@Component({
+  selector: 'vtm-technical-record-container',
+  template: `
+    <ng-container *ngIf="vehicleTechnicalRecord$ | async as vTechRecord">
+      <ng-container *ngIf="vTechRecord.techRecord | FilterRecord as activeRecord">
+        <vtm-technical-record
+          [vehicleTechRecord]="vTechRecord"
+          [activeRecord]="activeRecord"
+          [metaData]="metaData$ | async"
+          [editRecord]="viewState$ | async"
+          [testResultJson]="testResultJson$ | async"
+          (submitTechRecord)="onTechRecordSubmission($event)"
+          (changeViewState)="viewStateHandler($event)"
+        >
+        </vtm-technical-record>
+      </ng-container>
+    </ng-container>
+  `,
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class TechnicalRecordsContainer implements OnInit {
+  vehicleTechnicalRecord$: Observable<VehicleTechRecordModel>;
+  testResultJson$: Observable<TestResultModel>;
```

Comment:
In light of TechRecord/Adr implementation, I would suggest we make this amendment in a future ticket. Because doing it will increase the file size to be reviewed. Because changing to TestResultModel [] will cause some state files to change hence increase the workload maintaining while focusing on TechRecord/Adr